### PR TITLE
optimize the url

### PR DESCRIPTION
**Pull request by IndiraFinish**
[feature #768](https://github.com/apache/dubbo-go/issues/768)

### CODE REVIEW

**Review by zouyx — COMMENTED** [review]
Could you provide a test case for this change?

**Comment by IndiraFinish** [comment]
ok



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Joe Zou"<notifications@github.com&gt;; 
发送时间: 2020年9月21日(星期一) 晚上10:41
收件人: "apache/dubbo-go"<dubbo-go@noreply.github.com&gt;; 
抄送: "炮打资产阶级司令部"<2484713618@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [apache/dubbo-go] optimize the url (#769)





 
@zouyx commented on this pull request.
 
Could you provide a test case for this change?

**Review by beiwei30 — CHANGES_REQUESTED** [review]
should we use the same Set() for 'ConfigCenterConfig' and 'RegistryConfig'?

File: common/url.go [hunk 1]
```diff
@@ -99,6 +99,13 @@ type URL struct {
 	SubURL *URL
 }
```
Comment (watermelo) [inline common/url.go:101]: Maybe we should add some code comments.

**Comment by IndiraFinish** [comment]
我认为是的，他们都需要修改。其实最好的解决方式应该是在net.url.Set中解决，但是go的sdk没办法修改。从另一个角度来说，我认为不应该使用`url.Values`这个属性。如果go的sdk升级（例如Set方法的改变）是不是会对dubbo-go url模型产生影响那？应该使用自己封装的工具包，例如dubbbo

**Comment by IndiraFinish** [comment]
I've added a new comment to the `Set `func and test case in [#773](https://github.com/apache/dubbo-go/pull/773), and this PR can be close.

**Review by AlexStocks — APPROVED** [review]

**Comment by zouyx** [comment]
> I've added a new comment to the `Set `func and test case in [#773](https://github.com/apache/dubbo-go/pull/773), and this PR can be close. 
> 
> 
> 
> image
> 
> 

if useless ，you can close it

**Pull request closed without merging**